lib/cretonne/meta/cdsl/instructions.py

```diff
@@ -122,6 +122,10 @@ def __init__(self, name, doc, ins=(), outs=(), **kwargs):
                 i for i, o in enumerate(self.ins) if o.is_immediate())
 
         self._verify_polymorphic()
+        for attr in kwargs:
+            if not Instruction.ATTRIBS.has_key(attr):
+                raise RuntimeError(
+                        "unknown instruction attribute '" + attr + "'")
```

Comment:
I tend to use `AssertionError` for these things, but I don't know if it makes a difference.